daemon/api.go

```diff
@@ -801,11 +804,20 @@ func modeFlags(devMode, jailMode bool) (snapstate.Flags, error) {
 		if devMode {
 			return flags, errModeConflict
 		}
+		if classic {
+			return flags, errClassicJailmodeConflict
+		}
 		flags.JailMode = true
 	}
-	if devMode || devModeOS {
+	if devMode {
```

Comment:
Please note that last line. The original one in the code doesn't look right. If we do allow devMode, the classic setting should not clobber it.